php/php.editor/src/org/netbeans/modules/php/editor/completion/PHPCodeCompletion.java

```diff
@@ -2802,6 +2806,34 @@ public String extractUnqualifiedSuperClassName() {
             };
         }
 
+        static EnclosingClass forEnumDeclaration(final EnumDeclaration enumDeclaration) {
+            return new EnclosingClass() {
+                @Override
+                public String getClassName() {
+                    return enumDeclaration.getName().getName();
+                }
+
+                @Override
+                public Expression getSuperClass() {
+                    return null;
+                }
+
+                @Override
+                public List<Expression> getInterfaces() {
+                    return enumDeclaration.getInterfaes();
```

Comment:
Nitpick: just noticed typo, getInterfa[c]es().

Reply:
Yes, I also noticed it before. However, this is a method of the public packages. So, I didn't change it...
https://github.com/apache/netbeans/blob/a9f577647cdadff3ae7941ab80910dae00b47342/php/php.editor/src/org/netbeans/modules/php/editor/parser/astnodes/TypeDeclaration.java#L74-L81

Should we add `getInterface()` as a new method?
e.g. 
```java
    /**
     * List of interfaces that this type implements / extends.
     *
     * @return interfaces
     * @deprecated instead, use {@link #getInterfaces()}
     */
    @Deprecated
    public List<Expression> getInterfaes() {
        return getInterfaces();
    }

    /**
     * List of interfaces that this type implements / extends.
     *
     * @return interfaces
     */
    public List<Expression> getInterfaces() {
        return Collections.unmodifiableList(interfaces);
    }

```

Reply:
I see. Hmm, since it is only a friend API (is it, right?), we could perhaps do it and remove the deprecated method in 1 or 2 releases. What do you think?

Reply:
> since it is only a friend API (is it, right?),

Yes, right :)

> we could perhaps do it and remove the deprecated method in 1 or 2 releases. What do you think?

OK, I'll add the above change :) Thanks!

Reply:
Thanks a lot!

Reply:
Fixed https://github.com/apache/netbeans/pull/6873/commits/af05cd8aa6f16211b8749b8fd722cd1737efc0cb